### PR TITLE
CarminDatasetProcessing extension entity

### PR DESCRIPTION
**Pull request by KhalilKes**
This PR contains CarminDatasetProcessing entity extension for VIP context, code was tested in the development environment.

### CODE REVIEW

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+public class CarminDatasetPorcessingServiceImpl extends BasicEntityServiceImpl<CarminDatasetProcessing>  implements CarminDatasetProcessingService {
+
+        @Autowired
+        private CarminDatasetProcessingRepository carminDatasetProcessingRepository;
+
+        @Autowired
+        private ShanoirEventService eventService;
+
+        @Override
+        protected CarminDatasetProcessing updateValues(CarminDatasetProcessing from, CarminDatasetProcessing to) {
```
Comment (michaelkain) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java:29]: I would propose to make that method private please to better respect the principle of information hiding

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/dto/CarminDatasetProcessingDTO.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+
+/**
+ * DTO for CarminDatasetProcessing
+ * 
+ * @author KhalilKes
+ */
+public class CarminDatasetProcessingDTO extends DatasetProcessingDTO {
+
+    private String Identifier;
```
Comment (michaelkain) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/dto/CarminDatasetProcessingDTO.java:14]: All attributes shall by Java convention please start with lower case, thank you!

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                to.setOutputDatasets(from.getOutputDatasets());
+                to.setProcessingDate(from.getProcessingDate());
+                to.setStudyId(from.getStudyId());
+
+                return to;
+        }
+
+        @Override
+        public CarminDatasetProcessing createCarminDatasetProcessing(final CarminDatasetProcessing carminDatasetProcessing) {
+                ShanoirEvent event = new ShanoirEvent(ShanoirEventType.IMPORT_DATASET_EVENT,
```
Comment (michaelkain) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java:51]: I am not sure to require an event here, as we do not do any import here, we only save a json object to the database and do not process any images or files

Comment (michaelkain) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java:51]: probably all the event code could be removed here please

**Review by michaelkain — CHANGES_REQUESTED** [review]
Dear Khalil,
thank you a lot for your PR! It looks really promising.
Please find my comments below:
- Could you please provide a db changes script, that we can deploy your extension
  of the database as well in production? the same script shall be useful for Ilhem as
  well, as you already have a production too.
	- see examples under:          [shanoir-ng](https://github.com/fli-iam/shanoir-ng)/[docker-compose](https://github.com/fli-iam/shanoir-ng/tree/develop/docker-compose)/[database](https://github.com/fli-iam/shanoir-ng/tree/develop/docker-compose/database)/[db-changes](https://github.com/fli-iam/shanoir-ng/tree/develop/docker-compose/database/db-changes)/datasets/
- The DTO is an exact copy of the entity, I would propose to delete it and only use the entity? especially as you have already Json annotations on your ExecStatus in the model package. We could delete the mapper as well? I think it just copies the content.
- Furthermore I am wondering about the usage of private String resultsLocation; in carminDsProcessing?
- and please see my two minor direct comments
Thank you for your time and efforts and with kind regards,
Michael

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java (hunk 3 above)
Comment (KhalilKes) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java:51]: Okay, will do

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/dto/CarminDatasetProcessingDTO.java (hunk 2 above)
Comment (KhalilKes) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/dto/CarminDatasetProcessingDTO.java:14]: Thank you! 
I will do as you suggested below, removing the DTO and use the entity instead.

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java (hunk 1 above)
Comment (michaelkain) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java:29]: Hi Khalil, I had a second look and you can just leave it. Thank you!

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/dto/CarminDatasetProcessingDTO.java (hunk 2 above)
Comment (michaelkain) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/dto/CarminDatasetProcessingDTO.java:14]: Hi Khalil, I am waiting for your commit then. Thank you!

**Comment by michaelkain** [comment]
Dear Khalil, please write me an email, when you have integrated all above and want me to have another review. Thank you and with kind regards, Michael

**Comment by KhalilKes** [comment]
Dear Michael,

I adjusted the code as you requested from your last review, and tested in my env. 
Please check the sql script for the DB migration, open for your remarks.

with kind regards, Khalil.

**Comment by KhalilKes** [comment]
I almost forgot, about the resultsLocation, it's used to persist the result path we create in the front, simply put at the time we create a carminDSProcessing we create with a unique path (string), this path then is used to create the folder where the result will be. 
I hope my answer was clear enough, otherwise it'll make more sense after the front PR.

Thank you.

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/dto/CarminDatasetProcessingDTO.java (hunk 2 above)
Comment (michaelkain) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/dto/CarminDatasetProcessingDTO.java]: Perfect, thank you!

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java (hunk 3 above)
Comment (michaelkain) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/service/CarminDatasetPorcessingServiceImpl.java]: Perfect, thx!

**Review by michaelkain — APPROVED** [review]
Hi Khalil,
thank you for your good PR! I approve it by code review only, as I have no VIP instance to test it.
As discussed, we will test it finally on your pre-prod.
Thank you for your good work and we keep in touch,
Michael

**Pull request merged**